Backpack/OverlayView/Classes/BPKOverlayView.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2020 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+#import "BPKOverlayViewCornerStyles.h"
+#import "BPKOverlayViewOverlayTypes.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * `BPKOverlayView` is a subclass of `UIView` which allows you to apply Skyscanner style tints to background content.
+ */
+IB_DESIGNABLE @interface BPKOverlayView : UIView
+
+/**
+ * The corner-style to apply to the card.
+ */
+@property(nonatomic, assign) IBInspectable BPKOverlayViewCornerStyle cornerStyle;
+
+/**
+ * The kind of overlay to apply to the view.
+ */
+@property(nonatomic, assign) IBInspectable BPKOverlayViewOverlayType overlayType;
+
+/**
+ * The view to display in front of the tint.
+ */
+@property(nonatomic, strong, readonly) UIView *foregroundView;
```

Comment:
I'm thinking we should create these and expose them, but not allow consumers to set their own.

Reply:
Yeah 👍 Maybe make this one lazy though. Just like `contentView` and `backgroundView` in `UITableViewCell`.